src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -153,59 +151,59 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
             {
                 // Set some context on the CompletionItem so the CompletionResolveHandler can handle it accordingly.
                 result = SetResolveData(result.Value, serverKind);
-                if (IsSimpleImplicitExpression(documentSnapshot, request, serverKind))
+
+                if (serverKind == LanguageServerKind.CSharp)
                 {
-                    result = DoNotPreselect(result.Value);
-                    result = IncludeCSharpKeywords(result.Value, serverKind);
+                    result = MassageCSharpCompletions(request, documentSnapshot, result);
                 }
             }
 
             return result;
         }
 
-        private bool IsSimpleImplicitExpression(LSPDocumentSnapshot documentSnapshot, CompletionParams request, LanguageServerKind serverKind)
+        private SumType<CompletionItem[], CompletionList>? MassageCSharpCompletions(
```

Comment:
nit: the method signature of this is inconsistent with the other signatures. Some take DocumentSnapshot first others take it second. Normalizing it either way so it's consistent will make future reading/fixing easier.